doc/lightning-createonion.7.md

```diff
@@ -0,0 +1,126 @@
+lightning-createonion -- Low-level command to create a custom onion
+===================================================================
+
+SYNOPSIS
+--------
+
+**createonion** *hops* *assocdata* \[*session_key*\]
+
+DESCRIPTION
+-----------
+
+The **createonion** RPC command allows the caller to create a custom onion
+with custom payloads at each hop in the route. A custom onion can be used to
+implement protocol extensions that are not supported by c-lightning directly.
+
+The *hops* parameter is a JSON list of dicts, each specifying a node and the
+payload destined for that node. The following is an example of a 3 hop onion:
+
+```json
+[
+	{
+		"type": "legacy",
+		"pubkey": "022d223620a359a47ff7f7ac447c85c46c923da53389221a0054c11c1e3ca31d59",
+		"payload": "000067000001000100000000000003e90000007b000000000000000000000000"
+	}, {
+		"type": "legacy",
+		"pubkey": "035d2b1192dfba134e10e540875d366ebc8bc353d5aa766b80c090b39c3a5d885d",
+		"payload": "000067000003000100000000000003e800000075000000000000000000000000"
+	}, {
+		"type": "legacy",
```

Comment:
We should change "type" to "style" since we've used that elsewhere.  Consistency in APIs is really important.

I'd also be in favor of changing "pubkey" here to "id", though the former is more correct.